### ISSUE TITLE
Conan should let my recipe inherit its package name and version

### ISSUE THREAD

**Issue by thejohnfreeman** [conan-io/conan#5340]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Conan: 1.15.0
OS: Ubuntu 18.04
Python: 3.7.1

My recipe wants to inherit its package name and version from a class that knows how to get them, but Conan refuses to create the package because (I believe) it uses `astroid` to parse the recipe script and see if there is a `name` and `version` attribute set directly on the recipe class. Please just run the recipe and fail if there really is no name or version.

```python
# test.py
from conans import ConanFile

class Foo:
    name = 'foo'
    version = '1.2.3'

class Recipe(Foo, ConanFile):
    pass
```

```
$ conan create test.py demo/testing
ERROR: conanfile didn't specify name
$ python -c 'import test; print(test.Recipe.name); print(test.Recipe.version)'
foo
1.2.3
```

I shouldn't have to do this.

```python
# fixed.py
from conans import ConanFile

class Foo:
    name = 'foo'
    version = '1.2.3'

class Recipe(Foo, ConanFile):
    name = Foo.name
    version = Foo.version
```

**Comment by memsharded** [comment]
This has a reason: inheriting automatically the name and version of a base class was very confusing and problematic, because the main case of inheriting a base class is via ``python_requires``, and in that case, the name and version are the name and version of the python-require'd package, which is not what you want.

I think allowing this would be breaking behavior for those users and cannot be changed, unless until the python-requires issue (https://github.com/conan-io/conan/issues/4635) has been addressed (and no earlier than Conan 2.0, to be able to change this)

**Comment by memsharded** [comment]
```python
    def test(self):
        c = TestClient()
        conanfile = textwrap.dedent("""
            from conan import ConanFile

            class Foo:
                name = 'foo'
                version = '1.2.3'

            class Recipe(Foo, ConanFile):
                pass
                """)
        c.save({"conanfile.py": conanfile})
        c.run("create .")
```
Works in Conan 2.0
Closing ticket as solved
